### DISCUSSION TITLE
Can maplists (blocklists) have regex?

### DISCUSSION THREAD

**Discussion by flaviozluca** [rspamd/rspamd#4462] [Q&A]
Friends,

I have a blocklist of email addresses (text file), one email per line.

Can I use regex to specify a number used in the domain, as a spammer keeps changing his from address domains, but always preserving same base domain adding numbers?

Ex: spammer@mydomain1.com
spammer@mydomain2.com
etc.

Thanks.

**Comment by ManChicken1911** [comment]
Sure. Make sure the map type has `regexp=true` set and then format them like:

`/spammer@mydomain[0-9]+\.com/`

which would do one or more numbers 0-9. I often do things like:

`/(@|\.)mydomain[0-9]+\.com/`

which would block any address at @mydomain4.com or any subdomain(s) like @blah.mydomain4.com

**Comment by flaviozluca** [comment]
Very nice of you ManChicken! Thanks a lot!!

**Comment by flaviozluca** [comment]
Do I have to put the initial and ending slash?

I added 1 like this

sousanogueira.*.com.br

I have to add those slashes?

Thanks!

**Reply by ManChicken1911** [reply]
Yes. Also, since . is a regex special character (means 'any character') you probably want to escape the periods in the rest of the domain name to match actual periods, so you'd wind up with:

`/sousanogueira.*\.com\.br/`

.* means "any character (.) zero or more times (*)" so the above pattern would match strings like:

```
sousanogueira@somewhere.com.br
sousanogueira-is-spamming-you@somewhere.else.com.br
someone@sousanogueira.com.br
```

**Comment by flaviozluca** [comment]
Thanks!

souzanogueira.*.com.br would catch just the domain part, right? How would it differ from the one you sent last /sousanogueira.*\.com\.br/  ?

**Reply by ManChicken1911** [reply]
Well, I guess that depends on your multimap configuration and how you have it setup.. if the filter type is is "email:domain" then yes it'd only match against the domain name portion of the email address.

**Comment by flaviozluca** [comment]
Thanks!
